src/pages/index.js

```diff
@@ -81,35 +143,186 @@ profileEditButton.addEventListener("click", () => {
+
+/* ------------------------ Enable form validation ------------------------ */
+
+const editFormValidator = new FormValidator(
+  validationSettings,
+  profileEditForm
+);
+const addFormValidator = new FormValidator(
+  validationSettings,
+  addCardFormElement
+);
+
+const editAvatarValidator = new FormValidator(
+  validationSettings,
+  document.querySelector(".modal__avatar")
+);
 
 editFormValidator.enableValidation();
 addFormValidator.enableValidation();
-
+editAvatarValidator.enableValidation();
 /* -------------------------------------------------------------------------- */
 /*                                 Functions;                                 */
 /* -------------------------------------------------------------------------- */
-function handleImageClick(data) {
-  imagePopup.open(data);
+function handleCardPreview(cardData) {
+  previewImagePopup.open(cardData);
+}
+
+const avatarSubmitButton = document.querySelector("#avatar-submit-button");
+
+// function handleAvatarEditSubmit(input) {
+//   const link = input.avatar;
+//   api.updateAvatar(link);
+//   if (link) {
+//     avatarSubmitButton.textContent = "Saving...";
+//     api
+
+//       .editProfileImage(link)
+//       .then((userData) => {
+//         userInfo.setUserInfo(userData);
+//         editAvatarModal.close();
+//       })
+//       .catch((err) => console.error("Error updating avatar:", err))
+//       .finally(() => {
+//         editAvatarModal.setLoadingState(false);
+//       });
+//   } else {
+//     console.error("Avatar Link is not defined");
+//   }
+// }
+
+function handleAvatarEditSubmit(input) {
+  const link = input.avatar;
+
+  if (link) {
+    // set loading state to true
+    editAvatarModal.setLoadingState(true);
+    // the right api call is updateAvatar
+    api
+      .updateAvatar(link)
+      .then((userData) => {
+        // set avatar info
+        userInfo.setAvatarInfo(userData.avatar);
+
+        editAvatarModal.close();
+      })
+      .catch((err) => console.error("Error updating avatar:", err))
+      .finally(() => {
+        // set loading state to false
+        editAvatarModal.setLoadingState(false);
+      });
+  } else {
+    console.error("Avatar Link is not defined");
+  }
 }
+
 function handleProfileEditSubmit(formValues) {
-  userInfo.setUserInfo({
-    name: formValues.title,
-    about: formValues.card__description,
-  });
+  editProfileModal.setLoadingState(true);
+  api
+    .updateProfileInfo(formValues.title, formValues.card__description)
+    .then(() => {
+      userInfo.setUserInfo({
+        name: formValues.title,
+        about: formValues.card__description,
+      });
+    })
+    .catch((err) => console.error("Error updating Profile Info", err))
+    .finally(() => {
+      editProfileModal.setLoadingState(false);
+    });
+
+  // userInfo.setUserInfo({
+  //   name: formValues.title,
+  //   about: formValues.card__description,
+  // });
   editProfileModal.close();
```

Comment:
`editProfileModal.close();` should be in line 236. You need to close modals only in `then` blocks after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well. **Please, correct it everywhere in the code**